### ISSUE TITLE
Refactor every set.?uid to share code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7152]
|<img alt="Tv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Tv reported|
|-|-|
|Trac ID|trac#352|
|Type|enhancement|
|Created|2003-10-19 18:43:34Z|

<details><summary>Searchable metadata</summary>

```
trac-id__352 352
type__enhancement enhancement
reporter__Tv Tv
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1066589014000000 1066589014000000
changetime__1468880289962746 1468880289962746
version__ 
owner__Tv Tv
cc__exarkun cc__Tv
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Tv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Tv commented|
|-|-|

```
#!html
<pre>
twisted.internet.process.switch_uid() is pretty good, it should be refactored
out of there, renamed switchUid(), and all setuid etc. callers should use it.
There have been too many bugs in this area, and they aren't detected by unit
tests; maybe a common implementation would even allow unit testing with suitable
drivers?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
twisted.python.util.switchUID exists now.  There are only two other places in
Twisted which call setuid(): twisted/internet/app.py, which no one should be
using anymore, and twisted/internet/process.py, which switches up to root and
then uses setreuid() and setregid().  Should switchUID be extended to cover this
use-case as well?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

`twisted/internet/app.py` no longer calls setuid.  process.py still does, but it does something more complicated than what switchUID can handle, so that's fine.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset 1069e367aa66123814f7ec0f993a11e7b6c0b5a3
```
#!CommitTicketReference repository="" revision="1069e367aa66123814f7ec0f993a11e7b6c0b5a3"
Merge pull request #352 from scop/8625-scop-assertequal

Author: scop
Reviewer: alex, rodrigc
Fixes: #8625

Use assertEqual instead of deprecated assertEquals
```
